### PR TITLE
[ARCTIC-196] Port Flink read/write API to adaptive hive reader/writer

### LINKED ISSUE #196: [Flink] Port Flink read/write API to adaptive hive reader/writer
#40 have implemented parquet reader/writer for hive.

Flink have to port to new API to support hive reading/writting.

### PR DESCRIPTION
**Pull request by zstraw**
Port Flink read/write API to adaptive hive reader/writer. resolve #196.

## Why are the changes needed?

#40 have already implemented parquet reader/writer for hive.
Flink need to port reader/writer to new API to support hive adaptive table.

## Brief change log
  - *The flink 1.12 and 1.14 versions are affected.*
  - *Flink read UnKeyed table and write all types of table.*
  - *Port iceberg file writter in writting unkeyed table to Arctic file writter.*

## How was this patch tested?
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? yes
  - If yes, how is the feature documented? not documented

### CODE REVIEW

File: core/src/test/java/com/netease/arctic/listener/ArcticRunListener.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+
+public class ArcticRunListener extends RunListener {
+  private long startTime;
+
+  @Override
+  public void testRunStarted(Description description) {
+    startTime = System.currentTimeMillis();
+    System.out.println("Tests started! Number of Test case: " +
```
Comment (YesOrNo828) [inline core/src/test/java/com/netease/arctic/listener/ArcticRunListener.java:33]: Please use LOG.info().

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/ArcticFileWriter.java [hunk 2]
```diff
@@ -89,6 +89,7 @@ public ArcticFileWriter(
     this.submitEmptySnapshot = submitEmptySnapshot;
     LOG.info("ArcticFileWriter is created with minFileSplitCount: {}, upsert: {}, submitEmptySnapshot: {}",
         minFileSplitCount, upsert, submitEmptySnapshot);
+    LOG.info("upsert is {}", upsert);
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/ArcticFileWriter.java:92]: This is duplicated log.

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java [hunk 3]
```diff
@@ -38,12 +37,18 @@
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java:40]: Is this an unused import?

File: flink/v1.14/flink/src/main/java/com/netease/arctic/flink/write/FlinkTaskWriterBuilder.java [hunk 4]
```diff
@@ -60,7 +60,7 @@ public class FlinkTaskWriterBuilder implements TaskWriterBuilder<RowData> {
   private int partitionId = 0;
   private long taskId = 0;
   private RowType flinkSchema;
-  private int mask;
+  private long mask;
```
Comment (YesOrNo828) [inline flink/v1.14/flink/src/main/java/com/netease/arctic/flink/write/FlinkTaskWriterBuilder.java:63]: Why change to long primitive type?

File: flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java [hunk 5]
```diff
@@ -83,9 +82,6 @@
 
 public class FlinkTestBase extends TableTestBase {
   private static final Logger LOG = LoggerFactory.getLogger(FlinkTestBase.class);
-  @ClassRule
-  public static final MiniClusterWithClientResource MINI_CLUSTER_RESOURCE =
```
Comment (YesOrNo828) [inline flink/v1.12/flink/src/test/java/com/netease/arctic/flink/FlinkTestBase.java:87]: Why the MiniClusterWithClientResource is removed?

**Review by YesOrNo828 — COMMENTED** [review]
@zstraw  I left some comments.

File: flink/v1.14/flink/src/main/java/com/netease/arctic/flink/write/FlinkTaskWriterBuilder.java (hunk 4 above)
Comment (zstraw) [reply flink/v1.14/flink/src/main/java/com/netease/arctic/flink/write/FlinkTaskWriterBuilder.java:63]: There is originally long, and this method is invoked by the parameter long type everywhere.

**Review by YesOrNo828 — APPROVED** [review]

**Pull request merged**